internal/objectstore/remote.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package objectstore
+
+import (
+	"context"
+	"io"
+
+	"github.com/juju/worker/v4"
+	"github.com/juju/worker/v4/catacomb"
+
+	"github.com/juju/juju/core/objectstore"
+	"github.com/juju/juju/internal/errors"
+	"github.com/juju/juju/internal/worker/apiremotecaller"
+)
+
+// remoteFileObjectStore is a facade for the object store that uses a remote
+// worker to perform operations. The remoteFileObjectStore is a
+// TrackedObjectStore that ties the lifecycle of the objectStore and
+// remoteWorker together. To prevent additional complexity in the file
+// objectstore implementation, the file objectstore doesn't need to know how to
+// manage the remoteWorker.
+type remoteFileObjectStore struct {
+	catacomb catacomb.Catacomb
+
+	objectStore  TrackedObjectStore
+	remoteWorker worker.Worker
+}
+
+// newRemoteFileObjectStore returns a new remoteFileObjectStore.
+func newRemoteFileObjectStore(objectStore TrackedObjectStore, remoteWorker worker.Worker) (*remoteFileObjectStore, error) {
+	w := &remoteFileObjectStore{
+		objectStore:  objectStore,
+		remoteWorker: remoteWorker,
```

Comment:
Seems remoteWorker hasn't been used yet.

Reply:
Yes, it's just added to the catacomb init plan.